pkg/api/util/conditions.go

```diff
@@ -147,16 +148,20 @@ func SetCertificateCondition(crt *cmapi.Certificate, conditionType cmapi.Certifi
 	nowTime := metav1.NewTime(Clock.Now())
 	newCondition.LastTransitionTime = &nowTime
 
+	// Set the condition generation to be that of the given Certificate spec
+	newCondition.ObservedGeneration = crt.Generation
```

Comment:
👍 this certainly does make sense - though I am now slightly concerned that at some point in our codebase, someone may be calling this function and then generating a patch for the resource to update the API.

If someone does this, they'll be updating the observedGeneration *without* a guarantee that the observedGeneration is actually correct/there are no conflicts.

Perhaps we'd be better off creating a separate function setting a condition whilst also updating the observedGeneration so we can clearly document and make it clear in the function signature how the result of the function should be used.

WDYT?

Reply:
In theory, all of the consumers will be using this func to also set the observedGeneration, as this should  _always_  be set now. Let's perhaps add the generation as a parameter, then there is no ambiguity and people won't forget :+1: